embassy-stm32/src/can/fdcan.rs

```diff
@@ -199,9 +200,20 @@ impl<'d, T: Instance> CanConfigurator<'d, T> {
         Self {
             config,
             instance: FdcanInstance(peri),
+            properties: Properties::new(),
         }
     }
 
+    /// Get driver properties
+    pub fn properties(&self) -> &Properties<T> {
+        &self.properties
+    }
+
+    /// Get mutable driver properties
+    pub fn properties_mut(&mut self) -> &mut Properties<T> {
```

Comment:
do we really need the `properties/properties_mut` split? I think we can either
- Only make it `&mut`. 
- Only make it `&`, and make all methods in Properties use `&self` (even getters). Should be sound since setters only do volatile reg writes, don't modify data in `self`. And if we don't make it `Sync` you can't call `&self` methods concurrently from different threads.

Reply:
this is making the API more complicated for "ideological purity" reasons, not practical reasons. 

If you view `&`/`&mut` references as "shared" / "exclusive", and due to how the hardware works we can allow setting properties with a shared reference "for free" (no refcell/mutex overhead needed), then why not do it? It makes the API simpler for the user, easier to learn.

Reply:
Made changes according to the above, let me know what you think